### ISSUE TITLE
Support for native numeric enums

### ISSUE THREAD

**Issue by heivo** [asteasolutions/zod-to-openapi#43]
Hi,

first thank you for this awesome library!

I have a problem with native numeric enums, they are rendered as strings, e.g. when I do the following:

```ts
enum Test {
  First = 1,
  Second = 2
}

export const testSchema = z.object({
  test1: z.nativeEnum(Test),
  test2: z.nativeEnum(Test).openapi({example: Test.First}),
});
```

Then it is emitted as:
```json
{
  "test1": { "type": "string", "enum": ["First", "Second", 1, 2] },
  "test2": { "type": "string", "enum": ["First", "Second", 1, 2], "example": 1 }
}
```

Swagger UI renders this as
```json
{
  "test1": "First",
  "test2": "1"
}
```

Is there a simple fix or workaround for this?

**Comment by AGalabov** [comment]
Hello @heivo thank you for the kind wods.

Saddly, last time I looked into this issue I encountered a problem with the inner structure of a native numberic enum in zod's `ZodNativeEnum`. The problem in the code is described with a comment [here](https://github.com/asteasolutions/zod-to-openapi/blob/master/src/openapi-generator.ts#L613) 

We even have a skipped test for it [here](https://github.com/asteasolutions/zod-to-openapi/blob/master/spec/simple.spec.ts#L181)

I will try and take a second look at this soon. But for the time being I suggest you just add your documentation "manually" using the `.openapi` method. Hopefully that should resolve your issue for now and I'll get back to you when I have more information.
